### PR TITLE
build latest master

### PR DESCRIPTION
**Pull request by ocefpaf**
The `cmake_fixes.patch` seems to be applied upstream but this build failed for me locally.

### CODE REVIEW

**Comment by rsignell-usgs** [comment]
@ocefpaf, how best to proceed here?

**Comment by ocefpaf** [comment]
> @ocefpaf, how best to proceed here?

My last commit gets this to compile but locally I see a test failing. Let's wait for the CIs here.

**Comment by ocefpaf** [comment]
The failing tests is:

```
The following tests FAILED:
211 - nczarr_test_run_interop (Failed)
```

@WardF and @dopplershift is this something we should report upstream?

**Comment by dopplershift** [comment]
@ocefpaf I think it requires `unzip` to run the test?
```
2021-05-05T19:15:24.1611297Z 207/214 Test #212: nczarr_test_run_purezarr ..............   Passed    0.78 sec
2021-05-05T19:15:24.1611981Z test 213
2021-05-05T19:15:24.1612313Z         Start 213: nczarr_test_run_interop
2021-05-05T19:15:24.1612524Z 
2021-05-05T19:15:24.1613601Z 213: Test command: /bin/bash "-c" "export srcdir=$SRC_DIR/nczarr_test;export TOPSRCDIR=$SRC_DIR;$SRC_DIR/nczarr_test/run_interop.sh run_interop.sh "
2021-05-05T19:15:24.1614141Z 213: Test timeout computed to be: 1500
2021-05-05T19:15:24.3359418Z 213: $SRC_DIR/nczarr_test/run_interop.sh: line 42: unzip: command not found
2021-05-05T19:15:24.3376703Z 208/214 Test #213: nczarr_test_run_interop ...............***Failed    0.18 sec
2021-05-05T19:15:24.3377346Z $SRC_DIR/nczarr_test/run_interop.sh: line 42: unzip: command not found
2021-05-05T19:15:24.3377610Z 
```

**Comment by ocefpaf** [comment]
> @ocefpaf I think it requires `unzip` to run the test?

Ah. I thought that libzip was "it." Adding it now only as a test dep.

**Comment by WardF** [comment]
This should also be checked for by netcdf-c during configure time and either return an error message (which feels heavy handed) or disable the tests which use it (which feels not heavy-handed enough), if unzip isn't found on the system path.

**Comment by ocefpaf** [comment]
> This should also be checked for by netcdf-c during configure time and either return an error message (which feels heavy handed) or disable the tests which use it (which feels not heavy-handed enough), if unzip isn't found on the system path.

That would be nice. I have to confess that I prefer the heavy-handed approach when it comes to testing :smile:

**Comment by dopplershift** [comment]
Looks like the Windows failure is due to the lack of unzip (?!?) and m4.

osx_arm64 build is failing with: `/bin/sh: /Users/runner/miniforge3/conda-bld/libnetcdf_1620328684691/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/bin/m4: Bad CPU type in executable`

That last one seem like a packaging bug for m4?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -66,6 +66,7 @@ requirements:
     - {{ compiler('cxx') }}
     - {{ mpi }}  # [mpi == 'openmpi' and build_platform != target_platform]
   host:
+    - m4  # dev only
```
Comment (xylar) [inline recipe/meta.yaml:69]: I would move `m4` from `host` to `build`.  That's usually where it is in other packages and it fits with `make`, `cmake`, etc.  That might account for the platform issue you're seeing.

Comment (ocefpaf) [reply recipe/meta.yaml:68]: Indeed. Thanks!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -79,6 +82,7 @@ requirements:
     - hdf5 * {{ mpi_prefix }}_*
     - jpeg
     - libzip
+    - unzip
```
Comment (xylar) [inline recipe/meta.yaml:85]: https://github.com/conda-forge/unzip-feedstock/blob/master/recipe/meta.yaml#L14

Not available for Win :-(

Comment (ocefpaf) [reply recipe/meta.yaml]: I forgot the msys version of it. Thanks!

**Comment by ocefpaf** [comment]
Ouch I forgot to send this to the dev channel :-/ Another PR incoming....

**Comment by xylar** [comment]
Good catch.

**Comment by ocefpaf** [comment]
> Good catch.

Late catch you mean :smile: 

I cancelled all builds but drone. For some reason I do no have the rights to do that. I'll mark them as broken later.

**Comment by xylar** [comment]
Ah, I saw that you canceled most of them so I was hoping you'd caught it in time to save the hassle.  Sounds like mostly the case...

**Pull request merged**